Dockerfile

```diff
@@ -0,0 +1,35 @@
+# Use the official ROS 2 humble base image
+FROM ros:humble
+
+# Set the working directory
+WORKDIR /ros2_workspace/src
+
+COPY . .
+# Install additional dependencies
+RUN apt-get update \
+    && apt-get -y --quiet --no-install-recommends install \
+    gcc \
+    git \
+    libxml2-dev \
+    libxslt-dev \
+    python3 \
+    ros-humble-ament-cmake-clang-format\
+    ros-humble-rviz2
+
+# Set environment variables
+ENV ROS_DISTRO humble
```

Comment:
```suggestion
```

This is unecessary, and will get set up when you source the ROS environment a few lines down.